### ISSUE TITLE
Reports spurious error for RoundingMode enums in core macro `with-precision`

### ISSUE THREAD

**Issue by skylize** [clj-kondo/clj-kondo#1678]
**version**

2022.04.25

**macro usage**

Related to a macro provided by clojure.core: [`with-precision`](https://clojuredocs.org/clojure.core/with-precision)

**problem**

Clojure core includes a macro `with-precision` to set both precision and rounding mode for BigDecimal operations.

The macro optionally accepts an argument `:rounding`, which must be followed by one of `CEILING`, `FLOOR`, `HALF_UP`, `HALF_DOWN`, `HALF_EVEN`, `UP`, `DOWN`, and `UNNECESSARY`. The macro translates this to a matching enum constant from [`java.math.RoundingMode`](https://docs.oracle.com/javase/7/docs/api/java/math/RoundingMode.html) for use in the proceeding expression.

When used, Clj-kondo reports the enum names as "Unresolved symbol". They are not symbols. They are merely characters to be interpreted by the macro.

**repro**

--- *clj_kondo_with_precision.clj* ---
```clojure
(ns clj-kondo-with-precision)

;; Example copied from clojure core `master` branch:
;;   `test/clojure/test_clojure/vars.clj`: line: 45
(with-precision 6 :rounding CEILING (+ 3.5555555M 1))
```
on command line
```sh
$ clj-kondo --lint clj_kondo_with_precision.clj
clj_kondo_with_precision.clj:5:29: error: Unresolved symbol: CEILING
linting took 12ms, errors: 1, warnings: 0
```

**expected behavior**

Since this is a core macro, and all legal entries are fully enumerated in the docs, the linter should not require any special configuration to prevent reporting an error for any of the specified enums within the context of `with-precision`.

**Comment by borkdude** [comment]
Agreed, this should be fixed.
